subsys/secure_storage/Kconfig

```diff
@@ -0,0 +1,189 @@
+#
+# Copyright (c) 2018 - 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# Secure storage subsystem configuration options
+
+menuconfig SECURE_STORAGE
+	bool "Secure Storage"
+	depends on !BUILD_WITH_TFM
+	help
+	  The secure storage subsystem allows its users to store in a secure
+	  way data ensuring data integrity and confidentially by using AEAD
+	  algorithms. It supports several secure implementation back-ends to
+	  provide various levels of trust depending on the device security
+	  features.
+
+if SECURE_STORAGE
+module = SECURE_STORAGE
+module-str = secure_storage
+source "subsys/logging/Kconfig.template.log_config"
+
+
+menuconfig PSA_PROTECTED_STORAGE
+	bool "Protected Storage"
+	depends on SECURE_STORAGE
+	help
+	  The Protected Storage is designed to store in a secure
+	  way device intimate data and protects against physical attacks.
+	  It supports several storage back-ends and several secure
+	  implementation back-ends to provide various levels of trust.
+
+if PSA_PROTECTED_STORAGE
+
+config PSA_PROTECTED_STORAGE_PREFIX
+	string "Protected Storage object prefix"
+	depends on PSA_PROTECTED_STORAGE
+	default "ps/"
+	help
+	  This defines the prefix for Protected Storage object names.
+
+endif # PSA_PROTECTED_STORAGE
+
+menuconfig PSA_INTERNAL_TRUSTED_STORAGE
+	bool "Internal Trusted Storage"
+	depends on SECURE_STORAGE
+	default y
+	help
+	  The Internal Trusted Storage is designed to store assets that must
+	  be placed inside internal flash. Some examples of assets that require
+	  this are replay protection values for external storage and keys for
+	  use by components of the PSA Root of Trust.
+
+if PSA_INTERNAL_TRUSTED_STORAGE
+
+config PSA_INTERNAL_TRUSTED_STORAGE_PREFIX
+	string "Internal Trusted Storage object prefix"
+	depends on PSA_INTERNAL_TRUSTED_STORAGE
+	default "its/"
```

Comment:
I took that from the original design, which made this part of the interface, so it could be used by other implementations as well.
By moving it to the SECURE_STORAGE_STORAGE_SETTINGS_BACKEND part we can not distinguish between PS and ITS anymore which was whished at one point in this ever ongoing thing.

Reply:
Ok, I see that this has been put into the interface now, and is not easily change-able.

Maybe consider removing the "/" here and "." from suffixes, and let the storage backend form the path using slash and dots instead?
It just seems very specific to the settings backend, so I'm not sure if it is re-usable to others.

Reply:
Ok removed the / not sure where the . in the prefix symbols is?

Reply:
They're in the source file: 
```
#define SECURE_STORAGE_BACKEND_AEAD_FILENAME_SUFFIX_SIZE  ".size"
#define SECURE_STORAGE_BACKEND_AEAD_FILENAME_SUFFIX_FLAGS ".flags"
#define SECURE_STORAGE_BACKEND_AEAD_FILENAME_SUFFIX_DATA  ".data"
#define SECURE_STORAGE_BACKEND_AEAD_FILENAME_SUFFIX_NONCE ".nonce"
```

Reply:
Ah these, ok I was only looking in the Kconfigs

Reply:
Ok then We have to change the interface again to request dies data specify, 
So either work with a suffix or have its own function for it.